### ISSUE TITLE
Views using `build_nested_view` are slow

### ISSUE THREAD

**Issue by arconis2020** [arconis2020/smb_to_kodi#31]
Profiling reveals 2 items that can be improved:

1. Database queries. This is actually the second largest time consumer but the easiest item to fix. Instead of using the simple queries we have now that bring back more data than necessary, we can use something more targeted like a `values_list`.
2. `add_media_file` is the largest time sink, brought on mostly by the fact that it gets called so many times. Speeding it up or reducing the number of calls would be a great improvement.

**Comment by arconis2020** [comment]
This was a very interesting exercise in using ONLY the bare minimum of data for your purposes, and doing things as few times as possible. A few notes for the future:

1. Pathlib is great if you want to do complex things with paths, but because it instantiates each path as an object with many attributes and possible functions, it's orders of magnitude slower than simple string manipulation. According to profiling in cProfile, pathlib is the slowest way to do path work, and while os.path.{basename|dirname} is significantly faster, the fastest way to mess with paths in this program by far is simply using a string split function. Since this program runs in a Linux Docker container with SMB paths that will always be split with "/", this is a valid approach. Now I'm using ONLY what I need, and nothing more, making Python significantly faster.
2. Django model objects are expensive to instantiate, and Django database queries are MUCH faster if you use a `values_list` instead of a simple `.all()` query to return all objects in a library. I was able to shave 400ms from a 1.4s runtime just by making this change. Again, using ONLY what I needed (the "smb_path" string) instead of convenient objects made Python significantly faster.
3. By reducing the number of function calls to lxml elements (in one case using a better instantiation call, in another case combining `append()` calls into a single `extend()`), I was able to further reduce the runtime and complexity of the `add_media_file` function. Current testing on my laptop shows that the new function runs in 308ms as compared to the old value of 761ms.

In summary, I reduced the number of custom objects and replaced them with native Python data types like strings and tuples, I reduced the number of function calls to necessary objects, and I stopped using convenient path functions for possible compatibility I would never need, resulting in a net runtime improvement of 61% (in dev) for the `build_nested_view` function.
